src/TSMapEditor/UI/Windows/ScriptsWindow.cs

```diff
@@ -515,15 +581,49 @@ private void ListScripts()
         {
             lbScriptTypes.Clear();
 
-            foreach (var script in map.Scripts)
+            List<Script> sortedScripts = map.Scripts;
```

Comment:
There's a few things that could be improved here. I'll be adding a few suggestions, they make sense when viewed as a whole.

```suggestion
            IEnumerable<Script> sortedScripts = map.Scripts;
```